Mathlib/Algebra/Order/Disjointed.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2025 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Algebra.Order.PartialSups
+import Mathlib.Data.Nat.SuccPred
+import Mathlib.Order.Disjointed
+/-!
+# `Disjointed` for functions on a `SuccAddOrder`
+
+This file contains material excised from `Mathlib.Order.Disjointed` to avoid import dependencies
+from `Mathlib.Algebra.Order` into `Mathlib.Order`.
+
+## TODO
+
+Find a useful statement of `disjointedRec_succ`.
+-/
+
+open Order
+
+variable {α ι : Type*} [GeneralizedBooleanAlgebra α]
+
+section SuccAddOrder
+
+variable [LinearOrder ι] [LocallyFiniteOrderBot ι] [Add ι] [One ι] [SuccAddOrder ι]
+
+theorem disjointed_add_one [NoMaxOrder ι] (f : ι → α) (i : ι) :
+    disjointed f (i + 1) = f (i + 1) \ partialSups f i := by
+  simpa only [succ_eq_add_one] using disjointed_succ f (not_isMax i)
```

Comment:
Can you be consistent between `disjointed_succ` and `disjointed_add_one` on whether you take `[NoMaxOrder ι]` or `¬ IsMax i`?

Reply:
That was a deliberate choice. The overwhelmingly most important use case of `disjointed_add_one` is when  `ι` is `ℕ` or something order-isomorphic to it. I can't think of any interesting `SuccAddOrders` having a maximal element. But there certainly do exist `SuccOrder`s with a maximal element, such as `Fin n`. So making this consistent would either mean losing `Fin n`, or having to supply a tedious `¬ IsMax` side condition in every application to `ℕ`, neither of which seems desirable.